torch/onnx/utils.py

```diff
@@ -467,80 +489,81 @@ def _export(model, args, f, export_params=True, verbose=False, training=False,
-                                                                                         operator_export_type,
-                                                                                         f)
-        graph, params_dict, torch_out = _model_to_graph(model, args, verbose,
-                                                        training, input_names,
-                                                        output_names, operator_export_type,
-                                                        example_outputs, propagate,
-                                                        _retain_param_name, val_do_constant_folding,
-                                                        fixed_batch_size=fixed_batch_size)
-
-        # TODO: Don't allocate a in-memory string for the protobuf
-        defer_weight_export = export_type is not ExportTypes.PROTOBUF_FILE
-        if dynamic_axes is None:
-            dynamic_axes = {}
-        if custom_opsets is None:
-            custom_opsets = {}
-
-        _validate_dynamic_axes(dynamic_axes, model, input_names, output_names)
-
-        if export_params:
-            proto, export_map = graph._export_onnx(
-                params_dict, opset_version, dynamic_axes, defer_weight_export,
-                operator_export_type, strip_doc_string, val_keep_init_as_ip, custom_opsets,
-                val_add_node_names, val_use_external_data_format, model_file_location)
-        else:
-            proto, export_map = graph._export_onnx(
-                {}, opset_version, dynamic_axes, False, operator_export_type,
-                strip_doc_string, val_keep_init_as_ip, custom_opsets, val_add_node_names,
-                val_use_external_data_format, model_file_location)
-
-        if enable_onnx_checker and \
-           operator_export_type is OperatorExportTypes.ONNX and \
-           not val_use_external_data_format:
-            # Only run checker if enabled and we are not using ATEN fallback and
-            # large model format export in not enabled.
-            _check_onnx_proto(proto)
-
-        if export_type == ExportTypes.PROTOBUF_FILE:
-            assert(len(export_map) == 0)
-            with torch.serialization._open_file_like(f, 'wb') as opened_file:
-                opened_file.write(proto)
-        elif export_type in [ExportTypes.ZIP_ARCHIVE, ExportTypes.COMPRESSED_ZIP_ARCHIVE]:
-            import zipfile
-            compression = zipfile.ZIP_DEFLATED \
-                if export_type == ExportTypes.COMPRESSED_ZIP_ARCHIVE \
-                else zipfile.ZIP_STORED
-            with zipfile.ZipFile(f, 'w', compression=compression) as z:
-                z.writestr(ONNX_ARCHIVE_MODEL_PROTO_NAME, proto)
-                for k, v in export_map.items():
-                    z.writestr(k, v)
-        elif export_type == ExportTypes.DIRECTORY:
-            import os
-            if os.path.exists(f):
-                assert(os.path.isdir(f))
+        with set_training(model, training):
```

Comment:
Since this encapsulating most of the method body, we should consider a different name, such as `select_model_mode` or `select_training_vs_eval_mode`. The word `set` may give an impression that the mode is set for ever, and not the fact that it is only temporarily changed.

Reply:
Also, a brioef comment here (also) about what `set_training` is doing would be nice.